test/unit/mongo_client_options.test.js

```diff
@@ -192,6 +191,33 @@ describe('MongoOptions', function () {
     expect(options.writeConcern).has.property('wtimeout', 2);
   });
 
+  it('should ignore undefined and null values in the options object', function () {
+    const options = parseOptions('mongodb://localhost:27017/', {
+      maxPoolSize: null,
+      servername: undefined,
+      randomopt: null,
+      otherrandomopt: undefined
+    });
+
+    // test valid option key with default value
+    expect(options).to.have.property('maxPoolSize', 100);
+
+    // test valid option key without default value
+    expect(options).not.to.have.property('servername');
+
+    // test invalid option keys that are null/undefined
+    expect(options).not.to.have.property('randomopt');
+    expect(options).not.to.have.property('otherrandomopt');
+  });
+
+  it('should throw an error on unrecognized keys in the options object if they are defined', function () {
+    expect(() =>
+      parseOptions('mongodb://localhost:27017/', {
+        randomopt: 'test'
+      })
+    ).to.throw(MongoParseError, /randomopt.+not supported$/);
```

Comment:
```suggestion
expect(() =>
      parseOptions('mongodb://localhost:27017/', {
        randomopt: 'test'
      })
    ).to.throw(MongoParseError, 'option randomopt is not supported');

    expect(() =>
      parseOptions('mongodb://localhost:27017/', {
        randomopt: 'test',
        randomopt2: 'test'
      })
    ).to.throw(MongoParseError, 'options randomopt, randomopt2 are not supported');
```
I figure we might as well test the pluralization fixes here, if you agree. Otherwise, I think the current tests are good!

Reply:
No strong objections here; the thought was that testing precise error messages can be considered too restrictive because the exact wording should not matter and changing, say, the capitalization of the first letter or providing more context in the message shouldn't break the test; I personally don't think it's that much of a pain to update the test if you update the message, so if the rest of the team agrees, it's good with me :)